relentless/optimize/method.py

```diff
@@ -0,0 +1,49 @@
+import abc
+
+from .objective import ObjectiveFunction
+
+class Optimizer(abc.ABC):
+
+    @abc.abstractmethod
+    def optimize(self, objective):
+        pass
+
+class SteepestDescent(Optimizer):
+    def __init__(self, alpha, max_iter, abs_tol, mode):
```

Comment:
It doesn't do anything now, but just in case, you should call the parent constructor
```
super().__init__()
```